cvat/apps/annotation/annotation.py

```diff
@@ -83,7 +83,33 @@ def slice(self, start, stop):
         splitted_data = AnnotationIR()
         splitted_data.tags = copy.deepcopy(list(filter(is_frame_inside, self.tags)))
         splitted_data.shapes = copy.deepcopy(list(filter(is_frame_inside, self.shapes)))
-        splitted_data.tracks = copy.deepcopy(list(filter(lambda y: len(list(filter(is_frame_inside, y['shapes']))), self.tracks)))
+        splitted_data.tracks = []
+
+        for track_ in self.tracks:
+            if not list(filter(is_frame_inside, track_['shapes'])):
```

Comment:
I don't look how the function is used by for the slice function the condition looks wrong. Let's say we have a track with two shapes only. First shape is on 0 frame and second shape is on 100 frame (0 - start, and 100 - outside). Let's say the slice function is called with start=10 and stop=20.  Looks like the track will be filtered out.